src/uct/ib/base/ib_device.c

```diff
@@ -308,16 +150,16 @@ static ucs_status_t uct_ib_device_create(struct ibv_device *ibv_device,
     }
 
     /* Open verbs context */
-    ibv_context = ibv_open_device(ibv_device);
-    if (ibv_context == NULL) {
+    dev->ibv_context = ibv_open_device(ibv_device);
+    if (dev->ibv_context == NULL) {
         ucs_error("Failed to open %s: %m", ibv_get_device_name(ibv_device));
         status = UCS_ERR_IO_ERROR;
         goto err;
     }
 
     /* Read device properties */
-    IBV_EXP_DEVICE_ATTR_SET_COMP_MASK(&dev_attr);
-    ret = ibv_exp_query_device(ibv_context, &dev_attr);
+    IBV_EXP_DEVICE_ATTR_SET_COMP_MASK(&dev->dev_attr);
+    ret = ibv_exp_query_device(dev->ibv_context, &dev->dev_attr);
```

Comment:
if posible, need to use exp api to detect capabilities such as odp, dc, contig pages...